framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+    return [cpuTime]
+          
+  def locateMaterialInFile(self,numberOfMPI):
+    """
+      Finds the material names in a given INSTANT output file.
+      @ In, numberOfMPI, integer, number of MPI user-selected
+      @ Out, materialsDict, dictionary, dictionary listing the media treated in a given mpi output.
+              format: {MPI-0:{1:fuel1_1, 2:fuel1_5, 3:{fuel1_7}}, MPI-2:{1:fuel1_2, 2:fuel1_3, 3:fuel1_4, 4:fuel1_6}}
+    """
+    materialsDict = {}
+    for mpi in xrange(0,numberOfMPI):
+      count = 0
+      materialsDict['MPI-'+str(mpi)] = {}
+      mediaUsed = self.numberOfMediaUsed(mpi,numberOfMPI)
+      if self.phisicsRelap is True and mpi >= numberOfMPI - 1 : pass
+      else:
+        with open(os.path.join(self.workingDir, self.instantOutputFileMPI[mpi])) as outfile:
+          for line in outfile:
+            if re.search(r'Density spatial moment',line):
+              count = count + 1
+              matLine = filter(None, line.split(' '))
+              materialsDict['MPI-'+str(mpi)][count] = matLine[matLine.index('Material') + 1]
+              if count == mediaUsed:
+                break
+    return materialsDict
+   
+  def getDecayHeat(self,timeStepIndex,matchedTimeSteps):
+    """
+      Reads the main output file to get the decay heat.
+      @ In, timeStepIndex, integer, number of the timestep considered
+      @ In, matchedTimeSteps, list, list of time steps considered
+      @ Out, decayLabelList, list, list of the decay labels (under the format fuel|isotope)
+      @ Out, decayList, list, list of the decay heat values
+    """
+    materialList = []
+    decayLabelList = []
+    decayList = []
+    timeStepList = []
+    for mpi in xrange(0,len(self.instantOutputFileMPI)):
+      decayFlag = 0 
+      breakFlag = 0 
+      matLineFlag = 0 
+      materialCounter = 0
+      with open(os.path.join(self.workingDir, self.instantOutputFileMPI[mpi])) as outfile:
+        for line in outfile:
+          if re.search(r'Decay Heat computed',line) and mpi == 0:
+            self.isDecayHeatPrinted, self.decayHeatUnits = self.findDecayHeat(line)
+            if self.isDecayHeatPrinted is False:
+              decayLabelList = ['decayHeat']
+              decayList = 0
+              return  decayLabelList,decayList
+          if re.search(r'INDIVIDUAL DECAY HEAT BLOCK',line):
+            decayFlag = 1
+            materialCounter = materialCounter + 1
+          if re.search(r'CUMULATIVE DECAY HEAT BLOCK',line):
+            decayFlag = 0
+          if re.search(r'BURNUP OUTPUT',line):
+            breakFlag = 1
+          if decayFlag == 1 and breakFlag == 0:
+            line = re.sub(r'\n',r'',line)
+            decayLine = filter(None, line.split(' '))
+            if decayLine != []:
+              stringIsFloatNumber = self.isFloatNumber(decayLine)
+            if stringIsFloatNumber is True and decayLine != []:
+              if (float(decayLine[0]) == float(matchedTimeSteps[timeStepIndex])):
```

Comment:
fi you do not add any tolerance... it might be possible that for "numerics" differences in the construction of these lists this bool expression will not be true even if it should